pkg/cloud/aws/actuators/machine_scope.go

```diff
@@ -136,54 +143,54 @@ func (m *MachineScope) Close() {
 
 	latestMachine, err := m.storeMachineSpec(m.Machine)
 	if err != nil {
-		klog.Errorf("[machinescope] failed to update machine %q in namespace %q: %v", m.Machine.Name, m.Machine.Namespace, err)
+		m.Error(err, "failed to update machine")
 		return
 	}
 
 	_, err = m.storeMachineStatus(latestMachine)
 	if err != nil {
-		klog.Errorf("[machinescope] failed to store provider status for machine %q in namespace %q: %v", m.Machine.Name, m.Machine.Namespace, err)
+		m.Error(err, "failed to store machine provider status")
 	}
 }
 
 // MachineConfigFromProviderSpec tries to decode the JSON-encoded spec, falling back on getting a MachineClass if the value is absent.
-func MachineConfigFromProviderSpec(clusterClient client.MachineClassesGetter, providerConfig clusterv1.ProviderSpec) (*v1alpha1.AWSMachineProviderSpec, error) {
+func MachineConfigFromProviderSpec(clusterClient client.MachineClassesGetter, providerConfig clusterv1.ProviderSpec, log logr.Logger) (*v1alpha1.AWSMachineProviderSpec, error) {
```

Comment:
Can this function be made a method? Is it used anywhere else? If so, I'm not sure if the log lines should actually be in here or expect the caller would log.

Reply:
yes, this needs refactoring. I'll add it to the list of follow on issues.